dashboard/config/environments/test.rb

```diff
@@ -19,42 +19,27 @@
 
   is_ci = !!ENV.fetch('CI', nil)
 
-  # test environment should use precompiled, minified, digested assets like production,
-  # unless it's being used for unit tests.
-  ci_test = !!(ENV['UNIT_TEST'] || is_ci)
-
-  unless ci_test
-    # Compress JavaScripts and CSS.
-    # webpack handles js compression for us
-    # config.assets.js_compressor = :uglifier
-    # config.assets.css_compressor = :sass
-
-    # Version of your assets, change this if you want to expire all your assets.
-    config.assets.version = '1.0'
-
-    # Avoid loading all i18n files up front, which can significantly slow down initialization.
-    # Instead, it only loads i18n files that belong to the current locale.
-    config.i18n.backend = Cdo::I18n::LazyLoadableBackend.new(lazy_load: true)
-  end
-
   # In CI environments (ie, Drone), stub relevant AWS services (currently just SageMaker)
   # so we can run UI tests for our AI Chat (ie, Generative AI) lab.
   if is_ci
     config.stub_aichat_aws_services = true
   end
 
+  # Disable Rails.cache when running unit tests (also in Drone, not sure if this is strictly desired).
+  config.cache_store = :memory_store, {size: 64.megabytes} if ENV['UNIT_TEST'] || is_ci
+
+  # Avoid loading all i18n files up front, which can significantly slow down initialization.
+  # Instead, it only loads i18n files that belong to the current locale.
+  if ENV['LAZY_LOAD_TRANSLATIONS']
```

Comment:
Perhaps it would be better to add this to `config.yml.erb`, or we could combine it with the existing config `load_locales`:
https://github.com/code-dot-org/code-dot-org/blob/91d428a4776e194d5a467b214b672f5060065d9a/config.yml.erb#L381-L385
```suggestion
  unless CDO.load_locales
```

Reply:
cc @Hamms

Reply:
oh interesting, that might work! I think it'd be `unless load_locales`, right? 🙃 

We could match this potentially, without the env check?

https://github.com/code-dot-org/code-dot-org/blob/91d428a4776e194d5a467b214b672f5060065d9a/dashboard/config/initializers/fast_localization.rb#L4

Reply:
I believe it would also be `false` in CI/DDT by default, so it's not what we want

Reply:
Oh sorry I misread, are you suggesting a new option for `load_locales` (true/false/"lazy")? Yeah I'm pretty clueless on the implications of these, curious for Elijah's thoughts.

Reply:
Since the string `lazy` is a truthy value, `!'lazy'` will be the same as `!true` == `false`, so it's safe to include it as a third option.
https://github.com/code-dot-org/code-dot-org/blob/91d428a4776e194d5a467b214b672f5060065d9a/dashboard/config/initializers/fast_localization.rb#L4

Reply:
I would move this to `application.rb` so that it can be enabled in `development` as well

Reply:
I would also move this to application.rb so that it can be enabled in development as well

Reply:
Oh, I love the idea of making this a config setting rather than a flag or environment variable! That definitely fits better with the concept of only being enabled for local development, and I also like the idea of enabling it for local development as a whole rather than just for tests.

I would definitely prefer to avoid mixing types in these settings, though. It technically works, but it can make things harder to grok. I'd like to instead add a new config value `CDO.lazy_load_translations` that's also a boolean which defaults to false. I think we should then encourage all developers to enable both this and `CDO.load_locales` in their local environments and try to solicit some feedback on how lazily-loading everything compares to proactively-loading just English and Spanish. If it's as fast or faster for most people, we might be able to remove `load_locales` entirely.

Reply:
I'd like to get to a point where locales either always are available or the locale drop-down has limited options in development environments... it is rather surprising to newcomers that the locale won't be switched unless configured explicitly to do so. so a default where one of these is true... eventually.

Reply:
Does this approach give us a clear way to have lazy loading enabled only when running subsets of unit tests locally (which the env flag approach does, despite its ugliness)? Are we overcomplicating this? How bad would it be to just enable this in test environments generally (per @Hamms [comment here](https://github.com/code-dot-org/code-dot-org/pull/57661#pullrequestreview-1969651627)?)

Reply:
I'd say the config setting approach represents a pivot in intent; rather than focusing on the situation in which we're just running subsets of units tests, we're focusing on any situation in which we might want to sacrifice similarity to production in favor of faster initialization times, particularly for situations in which you might be iterating. Notably, this applies to both running subsets of unit tests and doing manual testing on a local server. I think it explicitly does _not_ apply to test environments in general, since for CI/CD tests we do want to mirror production and don't care about rapid iteration.

Reply:
Got it, thanks Elijah! Updated to along the lines I think of what y'all were suggesting. LMK what you think.